packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -918,13 +917,13 @@ export class Project extends ProjectBase {
         }),
       ],
     });
-    this.role.attachInlinePolicy(policy);
-
-    // add an explicit dependency between the EC2 Policy and this Project -
-    // otherwise, creating the Project fails,
-    // as it requires these permissions to be already attached to the Project's Role
-    const cfnPolicy = policy.node.findChild('Resource') as CfnResource;
-    project.addDependsOn(cfnPolicy);
+    if (policy) {
+      // add an explicit dependency between the VPC Policy and this Project -
+      // otherwise, creating the Project fails,
+      // as it requires these permissions to be already attached to the Project's Role
+      const cfnPolicy = policy.node.defaultChild as CfnResource;
+      project.addDependsOn(cfnPolicy);
```

Comment:
wouldn't it work to add the dependency on the `policy` object? I believe it should work.

Reply:
* `project.addDependsOn(policy)` doesn't compile (`Policy` is not a `CfnResource`)
* `this.node.addDependency(policy)` adds way too much dependencies, including a circular one between `policy` and itself

Reply:
What about `this.node.addDependency(cfnPolicy)`?

Reply:
> What about `this.node.addDependency(cfnPolicy)`?

Nope. Same as `this.node.addDependency(policy)` - a million dependencies, including a cyclical one.